app/src/main/java/app/unbound/android/Themes.kt

```diff
@@ -16,7 +23,63 @@ class Themes : Manager() {
         val isInRecovery = Unbound.settings.get("unbound", "recovery", false) as Boolean
 
         if (isEnabled && !isInRecovery) {
-            this.apply()
+            val updateTheme = Unbound.info.classLoader.loadClass("com.discord.theme.ThemeModule").getDeclaredMethod("updateTheme", String::class.java)
+            XposedBridge.hookMethod(updateTheme, object : XC_MethodHook() { // Runs on startup
+                override fun beforeHookedMethod(param: MethodHookParam) {
+                    Log.d("Unbound", "[Themes] Hooked updateTheme! : ${param.args[0]}")
+
+                    if (param.args[0] !in stockThemes) {
+                        //doesnt function if you enable directly after adding | should impl https://developer.android.com/reference/android/os/FileObserver to update addons
+                        val theme = getTheme(param.args[0] as String)
+                        if (theme == null) { param.result = null; return }
+                        themeType = theme.bundle.type?.asString
+
+                        Log.d("Unbound", "[Themes] ${param.args[0]} is $themeType")
+
+                        rawConstructor(theme.bundle.raw)
+                        hookSemantic(theme.bundle.semantic)
+
+
+                        /* Reimplement updater branch:
+                            android.app.Activity r4 = r3.getCurrentActivity()
+                            if (r4 == 0) goto L58 // return
+                            com.discord.theme.a r0 = new com.discord.theme.a
+                            r0.<init>()
+                            r4.runOnUiThread(r0)
+                        */
+                        // Not sure if this is necessary but stock updateTheme does and we are preventing its execution
+                        // a.run() does eventually call some updateUI functions so i assume its useful for live updating
+                        val a = Unbound.info.classLoader.loadClass("com.discord.theme.a")
+                        val constructor = a.getDeclaredConstructor(param.thisObject.javaClass)
+                        val runnable = constructor.newInstance(param.thisObject) as Runnable
+
+                        Activities.current.get()?.runOnUiThread(runnable)
+
+                        param.result = null // Don't run stock updateTheme(), will crash if it gets a custom theme id
+                        return
+                    }
+
+                    raw.clear() // Remove custom colouring
+                    semanticHooks.forEach { it.unhook() }
+                    themeType = null
+                }
+            })
+
+            val themeManager = Unbound.info.classLoader.loadClass("com.discord.theme.ThemeManager")
+            fun hookIsThemeMethods(methodName: String, expectedType: String) {
+                XposedBridge.hookMethod(themeManager.getDeclaredMethod(methodName), object : XC_MethodHook() {

```

Comment:
Could you a predicate for error prevention here incase getDeclaredMethod doesn't return anything